### PR TITLE
Story/ece651 43 footer system testing

### PR DESCRIPTION
**Pull request by Riya0001**
Test cases created for footer.

Test results location - Documents/General/Sprint 5/System Testing/ECE651-43

### CODE REVIEW

**Review by MareenaFr — APPROVED** [review]

**Comment by db8patel** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95)
> 
> [Bug](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=CODE_SMELL) [5 Code Smells](https://sonarcloud.io/project/issues?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95) No Coverage information [](https://sonarcloud.io/component_measures?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=kar12thik_Online-Food-Ordering-and-Delivery-System&pullRequest=95&metric=new_duplicated_lines_density&view=list)

@Riya0001 Please resolve these code smells.

File: src/__tests__/footer.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+afterAll(() => {
+  jest.clearAllMocks();
+});
+
+const store = configureStore({ reducer: rootReducer });
+
+describe("Wrapping Firebase For Footer Test", () => {
+  let firestore;
+  beforeAll(() => {
+    firestore = firebase.firestore();
```
Comment (kar12thik) [inline src/__tests__/footer.test.js:31]: Why are we initializing firestore for Footer?

File: src/__tests__/footer.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import { configureStore } from "@reduxjs/toolkit";
+import rootReducer from "../reducers/index";
+import { Provider } from "react-redux";
+import firebase from "firebase/compat/app";
+import "firebase/compat/firestore";
+import "firebase/compat/auth";
+
+window.scrollTo = jest.fn();
```
Comment (kar12thik) [inline src/__tests__/footer.test.js:16]: Do we need this ?

File: src/__tests__/footer.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-undef */
+import { render, screen, cleanup, act } from "@testing-library/react";
+import { BrowserRouter } from "react-router-dom";
+import Footer from "../components/Footer";
+import React from "react";
+import App from "../App";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
```
Comment (kar12thik) [inline src/__tests__/footer.test.js:8]: Remove unused imports.

File: src/__tests__/footer.test.js (hunk 2 above)
Comment (Riya0001) [reply src/__tests__/footer.test.js:16]: We need configstore,provider and rootReducer

**Pull request closed without merging**